daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/BCDParsers.scala

```diff
@@ -83,19 +80,15 @@ class BCDIntegerRuntimeLengthParser(
 ) extends PackedBinaryIntegerBaseParser(e)
   with HasRuntimeExplicitLength {
 
-  override def toBigInteger(num: Array[Byte]): JBigInteger = DecimalUtils.bcdToBigInteger(num)
-  override def toBigDecimal(num: Array[Byte], scale: Int): JBigDecimal =
-    DecimalUtils.bcdToBigDecimal(num, scale)
+  override def toNumber(num: Array[Byte]): JBigInteger = DecimalUtils.bcdToBigInteger(num)
```

Comment:
Here the return type is different. toNumber returns a JBigInteger.

I'm surprised this even works. JBigInteger must be a sub-type of JBigDecimal.